javascript/javascript3/week1/Weather_api_geoloc/app.js

```diff
@@ -0,0 +1,234 @@
+          
+            // see full list of required and optional parameters:
+            // https://opencagedata.com/api#forward
+          
+            var request = new XMLHttpRequest();
+            request.open('GET', request_url, true);
+          
+            request.onload = function() {
+              // see full list of possible response codes:
+              // https://opencagedata.com/api#codes
+          
+              if (request.status === 200){
+                // Success!
+                var data = JSON.parse(request.responseText);
+                weather.fetchWeather(data.results[0].components.city);
+              } else if (request.status <= 500){
+                // We reached our target server, but it returned an error
+          
+                console.log("unable to geocode! Response code: " + request.status);
+                var data = JSON.parse(request.responseText);
+                console.log('error msg: ' + data.status.message);
+              } else {
+                console.log("server error");
+              }
+            };
+          
+            request.onerror = function() {
+              // There was a connection error of some sort
+              console.log("unable to connect to server");
+            };
+          
+            request.send();  // make the request
+          },
+          getLocation: function() {
+            function success (data) {
+                geocode.reverseGeocode(data.coords.latitude, data.coords.longitude);
+            }
+            if (navigator.geolocation) {
+                navigator.geolocation.getCurrentPosition(success, console.error)
+            } else {
+                weather.fetchWeather("Aarhus");
+            }
+            
+          }
+}
+
+
+  document.querySelector(".search button").addEventListener("click", function () {
+    weather.search();
+  });
+  
+  document
+    .querySelector(".search-bar")
+    .addEventListener("keyup", function (event) {
+      if (event.key == "Enter") {
+        weather.search();
+       
+      }
+    });
+  
+    
+    geocode.getLocation();
+
+    
+    
+  
+  
+  
+  
+
+/*const express = require("express");
+const https = require("https");
+const app = express();
+
+app.get("/", function(req, res) {
+
+   
+    const url = "https://api.openweathermap.org/data/2.5/weather?q=copenhagen&appid=671ff5c18f7398c99dc62c7c7e9c4b6c&units=metric"
+    https.get(url, function(response) {
+        console.log(response);
+    })
+    res.send("Server is up and running.")
+})
+
+const axios = require('axios');
+
+axios.get("https://api.openweathermap.org/data/2.5/weather?q=copenhagen&appid=671ff5c18f7398c99dc62c7c7e9c4b6c&units=metric")
+  .then(response => {
+    console.log(response.data.url);
+    console.log(response.data.explanation);
+  })
+  .catch(error => {
+    console.log(error);
+  });
+
+
+app.listen(3000, function( {
+    console.log("Server is running on port 8080");
+}))
+*/
```

Comment:
good job of trying to do it but that is some node js and express js . and you will throw  them later in node module. 
i recommend you to remove this commentd code from this file because you don't need them amerge a clean file to your main branch

Reply:
Yes, at first I was trying to make it out with npm and node. I tried to check it with localserver 3000 and 8000, but it didn't work out) so i did it this way and I'm reading more about node and npm now, looks very interesting! :)

Reply:
I have deleted the commented out code